src/corelibs/U2Core/src/datatype/primers/PrimerDimersFinder.cpp

```diff
@@ -101,16 +99,20 @@ void BaseDimersFinder::fillResultsForCurrentIteration(const QByteArray &homologo
         QByteArray curArray;
         curArray.append(homologousBases.at(i));
         curArray.append(homologousBases.at(i + 1));
-        bool homologousRegionEnded = !energyMap.contains(curArray);
+        bool homologousRegionEnded = !ENERGY_MAP.contains(curArray);
         if (!homologousRegionEnded) {
-            freeEnergy += energyMap[curArray];
+            freeEnergy += ENERGY_MAP[curArray];
         }
-        if (homologousRegionEnded || i == homologousBases.size() - 2) {
+        bool reachedBasesEnd = i == homologousBases.size() - 2;
+        if (homologousRegionEnded || reachedBasesEnd) {
             if (freeEnergy < maximumDeltaG) {
                 maximumDeltaG = freeEnergy;
                 resHomologousRegion = homologousBases;
                 overlappingRegion.startPos = startPos;
-                overlappingRegion.length = i - startPos + 2;
+                overlappingRegion.length = i - startPos + 1;
+                if (!homologousRegionEnded && reachedBasesEnd) {
```

Comment:
'reachedBasesEnd' is always true if 'homologousRegionEnded' is false.
I would suggest moving this change into a separate commit, because 
1) This is a bug-fix with no tests.
2) This fix initiates a discussion: for example what will happen if 'homologousRegionEnded' is true and 'reachedBasesEnd' is true at the same time.  This decision is very unclear from the code only and requires at least a detailed comment when this situation is described.

Reply:
I can remove it from here, but it's a highly important fix for the "PCR primer design" feature. Dimer finding results are wrong without it.

Reply:
"'reachedBasesEnd' is always true if 'homologousRegionEnded' is false" - no, it's not. These two variables check statements, which are not intersect each other. Simple example:
`homologousBases` = "AAG  CTT", `overlapStartPos` = 0. There are a lot of iterations when `homologousRegionEnded` is `false` and `reachedBasesEnd` is `false` too.

Reply:
https://ugene.dev/tracker/browse/UGENE-7422